### PR TITLE
Attempt to fix parallel decoding

### PR DESCRIPTION
**Pull request by drgolem**
Added unit test TestDecodeTextParallel to demo an issue of decoding in multiple goroutines.
run test:
```sh
go test -race ./...
```

result:
```sh
WARNING: DATA RACE
Write at 0x00c0000aa0ee by goroutine 36:
  golang.org/x/text/encoding/unicode.(*utf16Decoder).Reset()
      /Users/val/goLand/pkg/mod/golang.org/x/text@v0.3.2/encoding/unicode/unicode.go:286 +0x57
  golang.org/x/text/encoding.(*Decoder).Reset()
      <autogenerated>:1 +0x49
  golang.org/x/text/transform.doAppend()
      /Users/val/goLand/pkg/mod/golang.org/x/text@v0.3.2/transform/transform.go:689 +0x90
  golang.org/x/text/transform.Bytes()
      /Users/val/goLand/pkg/mod/golang.org/x/text@v0.3.2/transform/transform.go:674 +0x8f
  golang.org/x/text/encoding.(*Decoder).Bytes()
      /Users/val/goLand/pkg/mod/golang.org/x/text@v0.3.2/encoding/encoding.go:57 +0x30
  github.com/bogem/id3v2/v2.decodeText()
```

this can be fixed by
```go
    //result, err := fromXEncoding.Decoder().Bytes(src)
    result, err := fromXEncoding.encoding.NewDecoder().Bytes(src)
```
This is just a proposal, please suggest better fix.

### CODE REVIEW

File: v2/encoding.go [hunk 1]
```diff
@@ -30,12 +30,15 @@ func (e Encoding) String() string {
 type xencodingWrapper struct {
 	decoder *xencoding.Decoder
 	encoder *xencoding.Encoder
+
+	encoding xencoding.Encoding
 }
```
Comment (bogem) [inline v2/encoding.go:35]: I suggest to complete delete `xencodingWrapper`, use just `xencoding.Encoding` and create new encoder/decoder every time we need to encode/decode.

xencodingWrapper is used to just micro-optimise performance but apparently it just introduces issues

File: v2/encoding_test.go [hunk 2]
```diff
@@ -25,6 +26,54 @@ func TestDecodeText(t *testing.T) {
...
+	var wg sync.WaitGroup
+
+	for _, tc := range testCases {
+		wg.Add(1)
+
+		tc := tc
+		go func() {
+			defer func() {
+				wg.Done()
+			}()
```
Comment (bogem) [inline v2/encoding_test.go:66]:
```suggestion
			defer wg.Done()
```

File: v2/encoding_test.go [hunk 3]
```diff
@@ -25,6 +26,54 @@ func TestDecodeText(t *testing.T) {
...
+		{[]byte{0x00, 0x48, 0x00, 0xE9, 0x00, 0x6C, 0x00, 0x6C, 0x00, 0xF6}, EncodingUTF16BE, "Héllö"},
+		{[]byte{0x00, 0x48, 0x00, 0xE9, 0x00, 0x6C, 0x00, 0x6C, 0x00, 0xF6}, EncodingUTF16BE, "Héllö"},
+	}
+
+	var wg sync.WaitGroup
+
+	for _, tc := range testCases {
+		wg.Add(1)
+
+		tc := tc
```
Comment (bogem) [inline v2/encoding_test.go:62]: Why do we need it?

File: v2/encoding_test.go [hunk 4]
```diff
@@ -25,6 +26,54 @@ func TestDecodeText(t *testing.T) {
 	}
 }
 
+func TestDecodeTextParallel(t *testing.T) {
```
Comment (bogem) [inline v2/encoding_test.go:29]: Please add a link to the issue:
```suggestion
// See https://github.com/bogem/id3v2/pull/74
func TestDecodeTextParallel(t *testing.T) {
```

**Review by bogem — COMMENTED** [review]
Hey @drgolem!
Super sorry for taking so much time to review your PR and thanks a lot for finding and fixing the issue!

File: v2/encoding_test.go (hunk 3 above)
Comment (drgolem) [reply v2/encoding_test.go:63]: to avoid this: https://github.com/golang/go/wiki/CommonMistakes

**Review by bogem — APPROVED** [review]
Looks awesome! Thank you for the contribution 👍

**Pull request merged**